https://issues.redhat.com/browse/ACM-5200 Adding tolerations for GitOps (#5053)
open-cluster-management/rhacm-docs: applications/gitops_config.adoc

[#register-gitops]
== Registering managed clusters to GitOps

Complete the following steps to register managed clusters to GitOps:

> . Create managed cluster sets and add managed clusters to those managed cluster sets. See the _Creating a ManagedClusterSet_ documentation for more information.

Reviewer: Discussion point: This is where it seems to me that we could use the link in the text. When I read it, here is what I'm thinking: 
1. It reads "for more information", but there really isn't any information provided. Maybe if it was, "See the _Creating a ManagedClusterSet_ documentation for the procedure," or similar.
2. There really isn't any indication that there is another section where I can find the link to this information. Should I find the doc in the navigation? This one might get clearer when people get used to our docs and we are fully consistent with it.

Not saying this is on you, but maybe we should try to discuss some criteria for these as a team.

Reviewer 2: Why is this going in app?

Reviewer 2: I think we need to make it clearer if this is an Application management function because the intro appears to me more like cluster lifecycle. (?)

Reviewer 3: The resource related to app is `GitOpsCluster`. The `GitOpsCluster` spec refers to a `Placement`. The `Placement` only works if there is a `ManagedClusterSet` and `ManagedClusterSetBinding` which are cluster lifecycle resources. 

Just providing some extra context. I don't know if it's a good idea to talk about cluster lifecycle resources here or not.

Reviewer 2: Right, I think this was placed since we have gitops content here, however, it is important @cadawson that we understand the concept of the function so we can write it well for the user...I am not sure this should be placed here but want to encourage a review of the playback and a quick Slack discussion or meeting with the dev team.

really with new content like this, a meeting with the dev team may need to be the first step since some of it crosses components.

Author: @swopebe thanks Brandi for calling this out. I agree that this might be better in the Clusters folder. I'll wait on confirmation before making this update  FYI @elgnay @rokej

Reviewer 4: Although this doc involves lots of cluster lifecycle resources, I still think application is the right place for it because the whole doc describes the necessary configuration we need to support gitops.

Author: @elgnay thanks for confirming. @swopebe as you mentioned before, we should probably consider adding a GitOps folder. So far GitOps is referenced in the Apps folder and Governance folder

Reviewer 2: @dockerymick ok, it's something that we need to carefully look at...app is day two by comparison. I will set up an arch meeting.

I am not sure if the suggestion was to add an anchor for the "Additional information" section (I thought I saw a comment on that) and I don't think we need to go down that route, if so.

Author: I am suggesting that we create a GitOps book. I added Additional resources section myself, I guess it was self-requested lol

Author: Recently updated the titles in the `gitops_config.adoc` file and `tolerations_config.adoc` file with the updates from @mikeshng [PR](https://github.com/mikeshng/rhacm-docs/pull/1/files). Resolving this convo

> . Create managed cluster set binding to the namespace where Red Hat OpenShift GitOps is deployed. See the `multicloud-integrations managedclustersetbinding` example, which binds to the `openshift-gitops` namespace. See the _Creating a ManagedClusterSetBinding resource_] documentation for more information. 

Reviewer: ```suggestion
. Create a managed cluster set binding to the namespace where Red Hat OpenShift GitOps is deployed. For an example of binding the managed cluster to the `openshift-gitops` namespace, see the `multicloud-integrations managedclustersetbinding` example. In the _Additional resources_ section, see _Creating a ManagedClusterSetBinding resource_ for more general information about creating a `ManagedClusterSetBinding`. 
```

Reviewer: Same question about how do I know that there is an additional resources section?

Author: @cadawson updated your suggestion.

> . In the namespace that is used in managed cluster set binding, create a `Placement` custom resource to select a set of managed clusters to register to an {ocp-short} GitOps operator instance. Use the `multicloud-integration` placement example. See _Using ManagedClusterSets with Placement_ for placement information. 

Reviewer: ```suggestion
. In the namespace that is used in managed cluster set binding, create a `Placement` custom resource to select a set of managed clusters to register to an {ocp-short} GitOps operator instance. Use the `multicloud-integration` placement example as a template. See _Using ManagedClusterSets with Placement_ for placement information. 
```

+
*Notes:* 
+
- Only {ocp-short} clusters are registered to an {ocp} GitOps operator instance, not other Kubernetes clusters.

> - In some unstable network scenarios, the managed clusters might be temporarily  placed in `Unavailable` state. See _Configuring placement tolerations for {product-title-short} and OpenShift GitOps_ for more details.

Reviewer 3: ```suggestion
- In some unstable network scenarios, the managed clusters might be temporarily placed in `Unavailable` state. See _Configuring placement tolerations for {product-title-short} and OpenShift GitOps_ for more details.
```

  placementRef:
    kind: Placement
    apiVersion: cluster.open-cluster-management.io/v1beta1
    name: all-openshift-clusters <1>
----

> <1> from the following sample that `placementRef.name` is `all-openshift-clusters`, and is specified as target clusters for the GitOps instance that is installed in `argoNamespace: openshift-gitops`. The `argoServer.cluster` specification requires the `local-cluster` value.

Reviewer: ```suggestion
<1> The `placementRef.name` value is `all-openshift-clusters`, and is specified as target clusters for the GitOps instance that is installed in `argoNamespace: openshift-gitops`. The `argoServer.cluster` specification requires the `local-cluster` value.
```

Reviewer: Where is the `argoServer.cluster` value in the sample? Is this something that used to be in the sample, but is not now?

Author: @cadawson the value is shown on Lines 42 and 43
